### ISSUE TITLE
Check username first on EmailUserProvider.php

### ISSUE THREAD

**Issue by JHGitty** [FriendsOfSymfony/FOSUserBundle#1963]
This issue is related to #1336.

For example we have 2 users:
1. User "Tom" with email "**tom@tom.com**"
2. User "**tom@tom.com**" with email "tom@gmail.com"

Which user should be selected if the input for username is 'tom@tom.com'?

You could do these things:
a) Allow usernames only.
b) Allow email addresses only.
c) Allow both and first look for email address.
**d) Allow both and first look for username.**

Currently you can do everything except the last. For the last way you currently need to create your own provider.

I think we should add an option or a second provider to allow to check the username first.

**Comment by azine** [comment]
+1

**Comment by damienalexandre** [comment]
I don't think this can be fixed correctly using two fields like the `username_email` provider do, even with an option you will still have one bad behavior, in one way or another.

Also the "c)" option does not exists currently.

**Comment by azine** [comment]
The situation with the current implementation has changed: see #2650
